src/HttpHeader.cc

```diff
@@ -759,6 +774,16 @@ HttpHeader::addEntry(HttpHeaderEntry * e)
 
     debugs(55, 7, this << " adding entry: " << e->id << " at " << entries.size());
 
+    // verify whether adding the entry will not exceed the maximum header size allowed
+    if (const auto max = maxLen()) {
+        const auto delta = e->length();
+        if (len + delta > max) {
+            debugs(55, DBG_IMPORTANT, "cannot add header entry " << e->id << " because the header length is becoming too large: " <<
```

Comment:
If there is no compelling reason to add a `DBG_IMPORTANT` record, let's not add it, especially for transaction-scope problems that the proxy admin cannot control (in most cases). In general, the exception catcher (rather than the thrower) should report the exception.